### PR TITLE
Make configuration.py Pylint compatible

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/configuration.py [hunk 1]
```diff
@@ -878,7 +929,8 @@ def as_dict(*args, **kwargs):   # noqa: D103
     return conf.as_dict(*args, **kwargs)
 
 
-def set(*args, **kwargs):   # noqa: D103
+def set(*args, **kwargs):
```
Comment (mik-laj) [inline airflow/configuration.py]: > airflow/configuration.py:932:0: W0622: Redefining built-in 'set' (redefined-builtin)

Still error here.

File: airflow/configuration.py [hunk 2]
```diff
@@ -540,73 +573,93 @@ def as_dict(
...
+            opt = self._get_secret_option(section, key)
+            if opt:
+                if not display_sensitive:
                     opt = '< hidden >'
+                if display_source:
+                    opt = (opt, 'secret')
                 elif raw:
                     opt = opt.replace('%', '%%')
+                config_sources.setdefault(section, OrderedDict()).update({key: opt})
+                del config_sources[section][key + '_secret']
```
Comment (turbaszek) [inline airflow/configuration.py]:
```suggestion
            if not opt:
                continue
            if not display_sensitive:
                opt = '< hidden >'
            if display_source:
                opt = (opt, 'secret')
            elif raw:
                opt = opt.replace('%', '%%')
            config_sources.setdefault(section, OrderedDict()).update({key: opt})
            del config_sources[section][key + '_secret']
```
WDYT?

Comment (potiuk) [reply airflow/configuration.py]: Right

**Comment by potiuk** [comment]
Updated. - should be green now.

**Review by kaxil — APPROVED** [review]

File: airflow/configuration.py [hunk 3]
```diff
@@ -540,73 +573,94 @@ def as_dict(
...
+            opt = self._get_secret_option(section, key)
+            if opt:
+                if not display_sensitive:
+                    opt = '< hidden >'
+                if display_source:
+                    opt = (opt, 'secret')
+                elif raw:
+                    opt = opt.replace('%', '%%')
+                config_sources.setdefault(section, OrderedDict()).update({key: opt})
+                del config_sources[section][key + '_secret']
```
Comment (turbaszek) [inline airflow/configuration.py:608]:
```suggestion
            if not opt:
                continue
            if not display_sensitive:
                opt = '< hidden >'
            if display_source:
                opt = (opt, 'secret')
            elif raw:
            opt = opt.replace('%', '%%')
            config_sources.setdefault(section, OrderedDict()).update({key: opt})
            del config_sources[section][key + '_secret']
```

**Pull request merged**